listing-rewards/program/src/listings/create/mod.rs

```diff
@@ -154,52 +215,55 @@ pub fn handler(
     listing.canceled_at = None;
     listing.reward_redeemed_at = None;
 
-    let seeds = &[
+    let reward_center_signer_seeds: &[&[&[u8]]] = &[&[
         REWARD_CENTER.as_bytes(),
         auction_house_key.as_ref(),
         &[reward_center.bump],
-    ];
-    let signer_seeds = &[&seeds[..]];
+    ]];
+
+    let auction_house_program = ctx.accounts.auction_house_program.to_account_info();
 
-    mpl_auction_house::cpi::auctioneer_sell(
-        ctx.accounts
-            .set_auctioneer_sell_ctx()
-            .with_signer(signer_seeds),
+    let create_listing_ctx_accounts = AuctioneerSell {
+        wallet: ctx.accounts.wallet.to_account_info(),
+        token_account: ctx.accounts.token_account.to_account_info(),
+        metadata: ctx.accounts.metadata.to_account_info(),
+        auction_house: ctx.accounts.auction_house.to_account_info(),
+        auction_house_fee_account: ctx.accounts.auction_house_fee_account.to_account_info(),
+        seller_trade_state: ctx.accounts.seller_trade_state.to_account_info(),
+        free_seller_trade_state: ctx.accounts.free_seller_trade_state.to_account_info(),
+        authority: ctx.accounts.authority.to_account_info(),
+        auctioneer_authority: ctx.accounts.reward_center.to_account_info(),
+        ah_auctioneer_pda: ctx.accounts.ah_auctioneer_pda.to_account_info(),
+        token_program: ctx.accounts.token_program.to_account_info(),
+        system_program: ctx.accounts.system_program.to_account_info(),
+        program_as_signer: ctx.accounts.program_as_signer.to_account_info(),
+        rent: ctx.accounts.rent.to_account_info(),
+    };
+
+    let create_listing_params = AuctioneerSellParams {
         trade_state_bump,
         free_trade_state_bump,
         program_as_signer_bump,
         token_size,
-    )?;
+    };
 
-    Ok(())
-}
+    let signer_required_keys = vec![
+        ctx.accounts.reward_center.key(),
+        ctx.accounts.wallet.key(),
```

Comment:
https://github.com/metaplex-foundation/metaplex-program-library/blob/master/auctioneer/program/src/sell/mod.rs#L170

Thinking we should just do the check for the reward_center as signer and let the downstream cpi call handle checking the wallet.

Reply:
I'm afraid that would not be enough. If you look here,

https://github.com/metaplex-foundation/metaplex-program-library/blob/e597868bf2da242edbe33a3e05cc269c48a4f377/auction-house/program/src/cancel/mod.rs#L97 

and here

https://github.com/metaplex-foundation/metaplex-program-library/blob/e597868bf2da242edbe33a3e05cc269c48a4f377/auction-house/program/src/bid/mod.rs#L183

the auctioneer_authority account is already being expected as Signer over the auction_house program, hence anchor will indeed infer them as Signer. Its the wallet account that requires the is_signer flag flipped to true, as thats the one Unchecked inside auction_house. 

Since the auctioneer program had the inclusion of auctioneer_authority, I've added it, but the primary check is the Wallet account @kespinola.